src/display/canvas.js

```diff
@@ -785,12 +785,12 @@ function getImageSmoothingEnabled(transform, interpolate) {
   const scale = Util.singularValueDecompose2dScale(transform);
   // Round to a 32bit float so that `<=` check below will pass for numbers that
   // are very close, but not exactly the same 64bit floats.
-  scale[0] = Math.fround(scale[0]);
-  scale[1] = Math.fround(scale[1]);
   const actualScale = Math.fround(
     OutputScale.pixelRatio * PixelsPerInch.PDF_TO_CSS_UNITS
   );
-  return scale[0] <= actualScale && scale[1] <= actualScale;
+  return (
+    Math.fround(scale[0]) <= actualScale && Math.fround(scale[1]) <= actualScale
+  );
```

Comment:
I've not tested it, but do we still need to use `Math.fround` with the scale-values now?
Given that `Util.singularValueDecompose2dScale` now returns a `Float32Array`, is the rounding perhaps implicitly done here?

Assuming that the above is correct, this can be simplified as follows.
```suggestion
    // `scale` is a Float32Array.
    return scale[0] <= actualScale && scale[1] <= actualScale;
  );
```

Reply:
Good idea.
I just tested:
```js
arr = new Float32Array(1); 
for (let i = 0; i < 1e10; i++) { 
 const x = Math.random() * 1000;
 arr[0] = x;
 if (Math.fround(x) !== arr[0]) {
  console.log(x);
  break;
 }
}; 
```
in Firefox and in Chrome, and it's good.